main/src/ca/uwaterloo/flix/language/phase/OccurrenceAnalyzer.scala

```diff
@@ -137,40 +138,42 @@ object OccurrenceAnalyzer {
     case Expression.Binary(sop, op, exp1, exp2, tpe, loc) =>
       val (e1, o1) = visitExp(exp1)
       val (e2, o2) = visitExp(exp2)
-      val o3 = combineAll(o1, o2)
+      val o3 = combineAllSeq(o1, o2)
       (OccurrenceAst.Expression.Binary(sop, op, e1, e2, tpe, loc), o3)
 
     case Expression.IfThenElse(exp1, exp2, exp3, tpe, loc) =>
       val (e1, o1) = visitExp(exp1)
       val (e2, o2) = visitExp(exp2)
       val (e3, o3) = visitExp(exp3)
-      val o4 = combineAll(o1, combineAll(o2, o3))
+      val o4 = combineAllBranch(o1, combineAllBranch(o2, o3))
       (OccurrenceAst.Expression.IfThenElse(e1, e2, e3, tpe, loc), o4)
 
     case Expression.Branch(exp, branches, tpe, loc) =>
-      var (e1, o1) = visitExp(exp)
-      var bs = Map[Symbol.LabelSym, OccurrenceAst.Expression]()
-      for ((sym, exp1) <- branches) {
-        val (e2, o2) = visitExp(exp1)
-        o1 = combineAll(o1, o2)
-        bs += (sym -> e2)
-      }
-      (OccurrenceAst.Expression.Branch(e1, bs, tpe, loc), o1)
+      val (e1, o1) = visitExp(exp)
+      var (o2, bs) = branches.foldLeft(Map[VarSym, Occur](), Map[LabelSym, OccurrenceAst.Expression]())((acc, b) => b match {
+        case (sym, exp1) =>
+          val (e2, o3) = visitExp(exp1)
+          val o4 = combineAllBranch(acc._1, o3)
+          val bs = acc._2 + (sym -> e2)
+          (o4, bs)
+      })
```

Comment:
I wonder if this would not be easier if we did `val (occurs1, branches1) = branches.map(...).unzip`? Are you familiar with `unzip`?